File: Scripts/Outros/ADS/Java/201/Aula24_11/src/Calculadora.java

public class Calculadora 
{
    int pilha=10;
    int numero;
    int numero1;
    int resultado;
    
    void recarregar()
    {
       pilha = 10; 
    }
    int somar()
    {
        if (pilha>0)
        {
            resultado = (numero+numero1);
            pilha--;
            return resultado;
        }
        else
        {
            System.out.println("Calculadora sem pilha");
            return 0;
        }
    }
    
    int subtrair()
    {
        if (pilha>0)
        {
            if (numero>numero1)
            {
                resultado = (numero-numero1);
                pilha--;
                return resultado;
            }
            if (numero<numero1)
            {
                resultado = (numero1-numero);
                pilha--;
                return resultado;
            }
            else
            {
                System.out.println("Calculadora sem pilha");
                return 0;
            }
        }
    }
    
    int multiplicar()
    {
        if (pilha>0)
        {
            resultado = (numero*numero1);
            pilha--;
            return resultado;
        }
        else
        {
            System.out.println("Calculadora sem pilha");
            return 0;
        }
    }
    
    int dividir()
    {
        if (pilha>0)
        {
            if (numero>numero1)
            {
                resultado = (numero/numero1);
                pilha--;
                return resultado;
            }
            if (numero<numero1)
            {
                resultado = (numero1/numero);
                pilha--;
                return resultado;
            }
            else
            {
                System.out.println("Calculadora sem pilha"); 
                return 0;
            }
        }   
    }
}

File: Scripts/Outros/ADS/Java/501/aulas/03.aul30/Calculadora/Calculadora.php
<?php
  /*
    basicamente é usado para nao criar
    um novo objeto a cada utilização dos metodos  
  */

  //criando classe
  class Calculadora
  {
    //criando metodos estaticos
      static function somar($v1, $v2)
      {
        //retornando a soma dos valores
          return $v1+$v2;
      }

      static function subtrair($v1, $v2)
      {
        //retornando a subtração dos valores
          return $v1-$v2;
      }

      static function multiplicar($v1, $v2)
      {
        //retornando a multiplicação dos valores
          return $v1*$v2;
      }

      static function dividir($v1, $v2)
      {
        //retornando a divisão dos valores
          return $v1/$v2;
      }
  }
?>
